### DISCUSSION TITLE
Promise is still pending when calling UnwrapIfPromise()

### DISCUSSION THREAD

**Discussion by ccase47** [sebastienros/jint#1955] [Q&A]
Hi all,

I have implemented promises/tasks in the way described in dicussion #935 - [(solution in this reply)](https://github.com/sebastienros/jint/discussions/935#discussioncomment-1116085) and have run into an issue with `context.PendingTaskCount` being equal to 0, but the completion value as a promise still has the state "pending". This is happening randomly and not consistently. Is there another way to make sure promises aren't pending when calling `UnwrapIfPromise()`? Any ideas as to why this might be happening?

Thanks!

Edited to add that I'm using 4.0.1.

**Comment by lahma** [comment]
This is something I don't want to start discussing 😅

Generally all task interop is experimental and doesn't map well between the environments. Have you tried setting `options.ExperimentalFeatures = ExperimentalFeature.TaskInterop`?

**Reply by ccase47** [reply]
Understandable! From the research I've done the past week while trying to solve this I can tell async is a bit of a pain.

Yes I have that set for the engine. And I was originally in 3.0.0-beta-2033 when I started getting this error and upgraded to 4.0.1 to see if that fixed anything.

**Reply by lahma** [reply]
While I'm not probably going to invest my time on this, I would still like know whether it was `3.0.0-beta-2033` that started to show a regression or was it broken from start and no earlier version was in use?
